src/containers/BooksForm.js

```diff
@@ -1,30 +1,72 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createBook } from '../actions/index';
 
-function BooksForm() {
-  const bookCategory = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
-  return (
-    <div className="container">
-      <div className="col-5 mx-auto">
-        <form className="form">
-          <div className="form-group">
-            <label htmlFor="title">
-              Title
-              <input type="text" className="form-control" name="title" id="title" />
-            </label>
-          </div>
-          <div className="form-group">
-            <label htmlFor="category">
-             Category
-              <select className="form-control">
-                {bookCategory.map(val => <option key={val}>{val}</option>)}
-              </select>
-            </label>
-          </div>
-          <button type="submit" className="btn btn-secondary">submit</button>
-        </form>
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.bookCategory = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+    this.titleInput = React.createRef();
+    this.categoryInput = React.createRef();
+
+    this.state = {
+      title: '',
+      category: this.bookCategory[0],
+    };
+  }
+
+  handleChange = () => {
+    this.setState({
+      title: this.titleInput.current.value,
+      category: this.categoryInput.current.value,
+    });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { title, category } = this.state;
+    const { createBook } = this.props;
+    createBook({ title, category });
+    this.titleInput.current.value = '';
+    this.categoryInput.current.value = this.bookCategory[0];
+  }
```

Comment:
- Your `handleSubmit` function is not reseting the state as stated [here](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896828/) 
![image](https://user-images.githubusercontent.com/46115725/72565262-17d01d00-3877-11ea-97d5-b30cb6c2ee8f.png)
![stateReset](https://user-images.githubusercontent.com/46115725/72565475-8f05b100-3877-11ea-8dfe-53af1a724092.gif)

- Also it allows the submission of empty books, which should not happen.

## See this:

Reply:
Oh snap, that was my silly mistake, I'll fix it.

Reply:
@jofer86 We think we have fixed everything, take a look :)